comp110-worksheet-6/OxoBoard.cs

```diff
@@ -6,71 +6,130 @@
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // Return the contents of the specified square.
+        public Mark GetSquare(int x, int y)
+        {
+            return board[x, y];
+        }
 
-		// If the specified square is currently empty, fill it with mark and return true.
-		// If the square is not empty, leave it as-is and return False.
-		public bool SetSquare(int x, int y, Mark mark)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // If the specified square is currently empty, fill it with mark and return true.
+        // If the square is not empty, leave it as-is and return False.
+        public bool SetSquare(int x, int y, Mark mark)
+        {
+            if (board[x, y] == Mark.None)
+            {
+                board[x, y] = mark;
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
 
-		// If there are still empty squares on the board, return false.
-		// If there are no empty squares, return true.
-		public bool IsBoardFull()
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // If there are still empty squares on the board, return false.
+        // If there are no empty squares, return true.
+        public bool IsBoardFull()
+        {
+            for (int i = 0; i < board.GetLength(0); i++)
+            {
+                for (int j = 0; j < board.GetLength(1); j++)
+                {
+                    if (board[i, j] == Mark.None)
+                    {
+                        return false;
+                    }
+                }
+            }
 
-		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
-		// Otherwise, return Mark.None.
-		public Mark GetWinner()
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+            return true;
+        }
 
-		// Display the current board state in the terminal. You should only need to edit this if you are attempting the stretch goal.
-		public void PrintBoard()
-		{
-			for (int y = 0; y < 3; y++)
-			{
-				if (y > 0)
-					Console.WriteLine("--+---+--");
+        // If a player has three in a row, return Mark.O or Mark.X depending on which player.
+        // Otherwise, return Mark.None.
+        public Mark GetWinner()
+        {
+            for (int i = 0; i < board.GetLength(0); i++)
+            {
+                for (int j = 0; j < board.GetLength(1); j++)
+                {
+                    if (board[i, j] != Mark.None) //Only look inside the cell if it's not empty.
+                    {
+                        int currI = i + 2, currJ = j + 2;
+                        int[] currSquare = new int[] { i, j };
+                        if (currI < 3 && currJ < 3)
```

Comment:
What are you intending to check here? I think this is just causing some checks to be skipped, hence the failed unit tests